src/panels/config/helpers/dialog-helper-detail.ts

```diff
@@ -277,7 +282,16 @@ export class DialogHelperDetail extends LitElement {
     this._submitting = true;
     this._error = "";
     try {
-      await HELPERS[this._domain].create(this.hass, this._item);
+      const createdEntity = await HELPERS[this._domain].create(
+        this.hass,
+        this._item
+      );
+      if (this._params?.dialogClosedCallback && createdEntity.id) {
+        this._params.dialogClosedCallback({
+          flowFinished: true,
+          entryId: `${this._domain}.${createdEntity.id}`,
```

Comment:
`entryId` is normally used for the entity registry entry. This should be called `entityId`